packages/clay-link/src/index.tsx

```diff
@@ -85,14 +116,21 @@ const ClayLink = React.forwardRef<HTMLAnchorElement, IProps>(
 				[`btn-${displayType}`]: displayType && !outline && !borderless,
 				[`btn-outline-${displayType}`]:
 					displayType && (outline || borderless),
+				[thickness ? fontWeights[thickness] : '']: thickness,
+				[`text-${fontSize}`]: fontSize,
+				'text-underline': underline,
 			};
 		} else {
 			classes = {
 				'link-monospaced': monospaced,
 				'link-outline': outline,
 				'link-outline-borderless': borderless,
-				[`link-${displayType}`]: displayType && !outline,
+				[displayType ? linkDisplayTypes[displayType] : '']:
+					displayType && !outline,
 				[`link-outline-${displayType}`]: displayType && outline,
+				[thickness ? fontWeights[thickness] : '']: thickness,
+				[`text-${fontSize}`]: fontSize,
```

Comment:
I think we could use the `Text` component here instead of declaring the class. This helps in the future when we have to make changes, the more we reuse the components the better.

Reply:
I've been looking at this and it looks like we would get some markup like:
```html
<span class="text-4">
    <a class="c-link link-primary" href="">Primary</a>
</span>
```

or

```html
<a class="c-link link-primary" href="">
    <span class="text-4">Primary</span>
</a>
```

It makes more sense to me if we can use `ClayLink` as a wrapper for `ClayText` where we can apply the `a` tag, `href`, `tabindex`, or whatever to `ClayText`. We won't have the extra span tag and the classes will all be on the `a` element.

Reply:
Oh yes that's true, it makes sense to me too I think we can work on that later and add an API of `as={Link}` in the Text. For now, let's go as it is.